keps/sig-scheduling/20190926-default-even-pod-spreading.md

```diff
@@ -0,0 +1,278 @@
+see-also:
+  - "/keps/sig-aaa/20190221-even-pods-spreading.md"
+---
+
+# Default Even Pod Spreading
+
+## Table of Contents
+
+<!-- toc -->
+- [Release Signoff Checklist](#release-signoff-checklist)
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+- [Proposal](#proposal)
+  - [User Stories](#user-stories)
+    - [Story 1](#story-1)
+    - [Story 2](#story-2)
+  - [Implementation Details/Notes/Constraints](#implementation-detailsnotesconstraints)
+  - [Risks and Mitigations](#risks-and-mitigations)
+- [Design Details](#design-details)
+  - [API](#api)
+  - [Default rules](#default-rules)
+  - [How user stories are addressed](#how-user-stories-are-addressed)
+  - [Implementation Details](#implementation-details)
+    - [In the metadata/predicates/priorities flow](#in-the-metadatapredicatespriorities-flow)
+    - [In the scheduler framework](#in-the-scheduler-framework)
+  - [Test Plan](#test-plan)
+  - [Graduation Criteria](#graduation-criteria)
+- [Implementation History](#implementation-history)
+- [Alternatives [optional]](#alternatives-optional)
+<!-- /toc -->
+
+## Release Signoff Checklist
+
+- [ ] kubernetes/enhancements issue in release milestone, which links to KEP (this should be a link to the KEP location in kubernetes/enhancements, not the initial KEP PR)
+- [ ] KEP approvers have set the KEP status to `implementable`
+- [ ] Design details are appropriately documented
+- [ ] Test plan is in place, giving consideration to SIG Architecture and SIG Testing input
+- [ ] Graduation criteria is in place
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+## Summary
+
+With `EvenPodsSpreading`, workload authors can define spreading rules for their loads based on the
+topology of the clusters. 
+
+By introducing configurable default spreading rules, workloads can be spread in the topology of
+the cluster according to opinionated rules set by a cluster operator, without having to explicitly
+define them. And workloads with specific needs can still override the rules by defining them in
+their PodSpec.
+
+## Motivation
+
+In order for a workload (pod) to use `EvenPodsSpreading`:
+
+1. Authors have to have an idea of the underlying topology.
+1. PodSpecs become less portable if their spreading rules are tailored to a specific topology.
+
+On the other hand, cluster operators know the underlying topology of the cluster, which makes
+them suitable to provide default spreading rules for all workloads in their cluster.
+
+### Goals
+
+- Cluster operators can define default spreading rules for pods that don't provide any.
+- Workloads are spread with the default rules if they belong to the same service, controller,
```

Comment:
"controller" is vague, I checked the node and it here means "replication controller".

Reply:
yeah, controller refers to replication controller, which is the predecessor of ReplicaSet.